extensions/google-drive/src/utils.ts

```diff
@@ -58,11 +66,14 @@ export const getDirectories = (path: PathLike): Array<PathLike> =>
     .filter(isDirectory)
     .filter((dir) => !IGNORED_DIRECTORIES.includes(basename(dir)));
 
-export const saveFilesInDirectory = (path: PathLike, db: Database) =>
+export const saveFilesInDirectory = (path: PathLike, db: Database) => {
+  const preferences = getPreferenceValues<Preferences>();
+  const excludePaths = getExcludePaths();
   readdirSync(path).forEach((file) => {
     const filePath = join(path.toLocaleString(), file);
 
-    if (!isFile(filePath)) return;
+    if (!preferences.shouldShowDirectories && !isFile(filePath)) return;
+    if (excludePaths.includes(filePath)) return;
```

Comment:
Should `getDirectories()` filter `excludePaths` instead?

Reply:
No. `saveFilesInDirectory` inserts all files and directories in the passed path, so without this filter, exclude path would show in the results.